src/Command/Service/CreateCoreArchCommandService.php

```diff
@@ -0,0 +1,310 @@
+                    . $this->repositoryInterfaceTemplate->name . '</info>'
+            );
+        }
+        $output->writeln('<comment>' . $this->getRelativeFilePath($filePath) . '</comment>');
+        $output->writeln('');
+    }
+
+    /**
+     * @param OutputInterface $output
+     * @param FileTemplate $fileTemplate
+     */
+    public function createUnitTestFileIfNotExists(OutputInterface $output, FileTemplate $fileTemplate): void
+    {
+        $className = $fileTemplate->name . 'Test';
+        $filePath = $this->srcPath . '/../tests/php' . DIRECTORY_SEPARATOR
+            . preg_replace('/\\\\/', DIRECTORY_SEPARATOR, $fileTemplate->namespace)
+            . DIRECTORY_SEPARATOR . $className . '.php';
+        if (! file_exists($filePath)) {
+            preg_match('/^(.+).' . $className . '\.php$/', $filePath, $matches);
+            $dirLocation = $matches[1];
+            // Create dir if not exists,
+            if (! is_dir($dirLocation)) {
+                mkdir($dirLocation, 0777, true);
+            }
+            file_put_contents(
+                $filePath,
+                (new UnitTestTemplate())->generateContentForUnitTest('Test\\' . $fileTemplate->namespace)
+            );
+            $output->writeln(
+                '<info>Creating Test file : ' . 'Test\\' . $fileTemplate->namespace . '\\'
+                    . $className . '</info>'
+            );
+        } else {
+            $output->writeln(
+                '<info>Using Test file : ' . 'Test\\' . $fileTemplate->namespace . '\\'
+                . $className . '</info>'
+            );
+        }
+        $output->writeln('<comment>' . $this->getRelativeFilePath($filePath) . '</comment>');
+        $output->writeln('');
+    }
+
+    /**
+     * @param string $filePath
+     *
+     * @return string
+     */
+    public function getRelativeFilePath(string $filePath): string
+    {
+        return substr($filePath, strpos($filePath, self::REPO_PREFIX));
+    }
+
+    /**
+     * @return string
+     */
+    public function getSrcPath(): string
+    {
+        return $this->srcPath;
+    }
+
+    /**
+     * @return RepositoryInterfaceTemplate
+     */
+    public function getRepositoryInterfaceTemplate(): RepositoryInterfaceTemplate
+    {
+        return $this->repositoryInterfaceTemplate;
+    }
+
+    /**
+     * Look for existing model with the same name.
+     *
+     * @param string $modelName
+     *
+     * @return array<string,string>
+     */
+    private function searchExistingModel(string $modelName): array
+    {
+        // Search for all model with the same name.
+        $modelsInfos = iterator_to_array(
+            new \GlobIterator(
+                $this->srcPath . DIRECTORY_SEPARATOR . 'Core' . DIRECTORY_SEPARATOR
+                    . preg_replace('/New/', '', $modelName) . DIRECTORY_SEPARATOR . 'Domain' . DIRECTORY_SEPARATOR
+                    . 'Model' . DIRECTORY_SEPARATOR . $modelName . '.php'
```

Comment:
Indentation + simpler `str_replace`
```suggestion
                . str_replace('New', '', $modelName) . DIRECTORY_SEPARATOR . 'Domain' . DIRECTORY_SEPARATOR
                . 'Model' . DIRECTORY_SEPARATOR . $modelName . '.php'
```